src/sync/Synchroniser.js

```diff
@@ -56,52 +57,77 @@ export class Synchroniser {
     // initialisation, in which case we are continuing a failed partial initialisation. If the
     // serverURL is different, it is either completely fresh, or the URL has been changed so we
     // should start afresh
-    const isFresh = serverURL !== this.serverURL;
-    if (isFresh) this.database.write(() => { this.database.deleteAll(); });
+    const oldUrl = this.settings.get(SYNC_URL);
+    const isFresh = !oldUrl || serverURL !== oldUrl;
+    if (isFresh) {
+      this.database.write(() => {
+        this.database.deleteAll();
+      });
+    }
     try {
       await this.authenticator.authenticate(serverURL, syncSiteName, syncSitePassword);
       const thisSiteId = this.settings.get(SYNC_SITE_ID);
       const serverId = this.settings.get(SYNC_SERVER_ID);
-      if (isFresh) { // If a fresh initialisation, tell the server to prepare required sync records
+      if (isFresh) {
+        // If a fresh initialisation, tell the server to prepare required sync records
         await fetch(
           `${serverURL}/sync/v2/initial_dump/?from_site=${thisSiteId}&to_site=${serverId}`,
           {
             headers: {
               Authorization: this.authenticator.getAuthHeader(),
             },
-          });
+          }
+        );
         // If the initial_dump has been successful, serverURL is valid, and should now have all sync
         // records queued and ready to send. Safe to store as this.serverURL
         this.serverURL = serverURL;
       }
       await this.pull(setProgress);
-    } catch (error) { // Did not authenticate, sync error, or no internet, pass error up
+    } catch (error) {
+      // Did not authenticate, sync error, or no internet, pass error up
       throw error;
     }
     this.settings.set(SYNC_IS_INITIALISED, 'true');
     this.syncQueue.enable(); // Begin the sync queue listening to database changes
   }
 
   /**
-   * Return whether the synchroniser has been completely or partially initialised.
-   * @return {string} Either 'complete', 'partial', 'uninitialised'
+   * Return whether the synchroniser has been initialised.
+   * @return {boolean} True if initial sync has been completed successfully
    */
   isInitialised() {
     const syncIsInitialised = this.settings.get(SYNC_IS_INITIALISED);
     return syncIsInitialised && syncIsInitialised === 'true';
   }
 
   /**
+   * Return whether the whether or not the app is syncing. Useful for checking if sync was
+   * interrupted on app close or crash
+   * @return {boolean} 'true' if the app was (in case of app crash or close) or currently is syncing
+   */
+  isSyncing() {
```

Comment:
not a fan of having isSyncing in different places with similar meaning.  MSupplyMobileAppContainer, this.state.isSyncing, after site login in and fully synced initial dump. And here, if site is syncing, during or after initial dump  Maybe we can also have two callbacks to set MSupplyMobileAppContainer.state.isSyncing, at the same time this.setting.set(SYNC_IS_SYNCING.. or leave for now but we all take not for refactor